packages/mobx-react/src/observerClass.ts

```diff
@@ -165,6 +191,11 @@ function createReactiveRender(originalRender: any) {
         isRenderingPending = false
         // Create reaction lazily to support re-mounting #3395
         const reaction = (reactiveRender[mobxAdminProperty] ??= createReaction())
+
+        if (!this[reactionTrack]) {
+            addReactionToTrack(this, reaction, {})
```

Comment:
Move at the end of `createReaction`, otherwise you would have to clear `this[reactionTrack]` whenever `reactiveRender[mobxAdminProperty]` is cleared.